gpii/node_modules/contextManager/src/ContextManagerUtilities.js

```diff
@@ -0,0 +1,105 @@
+
+};
+
+// Transforms required for condition evaluation:
+fluid.defaults("gpii.contextManager.transforms.inRange", {
+    gradeNames: "fluid.standardTransformFunction"
+});
+
+gpii.contextManager.transforms.inRange = function (value, transformSpec) {
+    return (transformSpec.min && transformSpec.min > value) ||
+        (transformSpec.max && transformSpec.max < value) ? false : true;
+};
+
+// Transforms required for condition evaluation:
+fluid.defaults("gpii.contextManager.transforms.timeInRange", {
+    gradeNames: "fluid.standardTransformFunction"
+});
+
+
+gpii.contextManager.transforms.timeInRange = function (value, transformSpec) {
+    var isEarlier = gpii.contextManager.transforms.timeInRange.isEarlier;
+    var currentTime = {
+        hours: (new Date(value)).getHours(),
+        minutes: (new Date(value)).getMinutes()
+    };
+    var fromTime = gpii.contextManager.transforms.timeInRange.timeParser(transformSpec.from);
+    var toTime = gpii.contextManager.transforms.timeInRange.timeParser(transformSpec.to);
+
+    if (isEarlier(toTime, fromTime)) { //if time wraps
+        return (isEarlier(fromTime, currentTime) || isEarlier(currentTime, toTime));
+    } else {
+        return (isEarlier(fromTime, currentTime) && isEarlier(currentTime, toTime));
+    }
+};
+
+gpii.contextManager.transforms.timeInRange.timeParser = function (time) {
+    time = time.split(":");
+    return {
+        hours: parseInt(time[0], 10),
+        minutes: parseInt(time[1], 10)
+    };
+};
+
+// checks whether 'first' is earlier than 'second' - returns true if so
+gpii.contextManager.transforms.timeInRange.isEarlier = function (first, second) {
+    return (60 * (first.hours - second.hours) + first.minutes - second.minutes) < 0;
+};
+
+gpii.contextManager.utils.findActiveContexts = function (currentContext, matchData) {
+    var activeContexts = [ "gpii-default" ];
+    // do nothing if there is no NP set yet
+    fluid.each(matchData.inferredConfiguration, function (content, contextId) {
+        if (content.conditions === undefined) {
+            return;
+        }
+        // massage conditions to be proper transform specs:
+        var condArray = [];
+        fluid.each(content.conditions, function (condition) {
+            var entry = fluid.copy(condition);
+            entry.type = gpii.contextManager.utils.transformTypeLookup[condition.type];
+            if (entry.type === undefined) {
+                console.log("ERROR: Failed to find the condition type: " + condition.type + " - Ignoring that conditional block");
+                condArray = false;
+                return;
+            }
+            condArray.push({
+                transform: entry
+            });
+        });
+        var result = fluid.model.transformWithRules(currentContext, { evaluations: condArray });
+        // if we only have 'true' entries after evaluation
+        if (result.evaluations && $.inArray(false, result.evaluations) === -1) {
```

Comment:
I think we can now safely abandon the use of $.inArray and apply Array.indexOf instead - https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf - it's supported on all our now supported platforms.